### ISSUE TITLE
训练一段时候后挂掉。像是训练正常，预测出现问题。报错PyEval_EvalFrameEx相关。

### ISSUE THREAD

**Issue by liushanshan07** [PaddlePaddle/Paddle#14360]
Saved model to: ./output/models/model_186000.

Time: 1541876304.23; Iter[187000]; Avg Warp-CTC loss: 1.017; Avg seq err: 0.138

Time: 1541876684.59; Iter[188000]; Avg Warp-CTC loss: 0.914; Avg seq err: 0.131
*** Aborted at 1541876972 (unix time) try "date -d @1541876972" if you are using GNU date ***
PC: @                0x0 (unknown)
*** SIGSEGV (@0x22) received by PID 802 (TID 0x7fd87e73b700) from PID 34; stack trace: ***
    @     0x7fd87dd3c500 (unknown)
    @     0x7fd87e03cd5b PyEval_EvalFrameEx
    @     0x7fd87dfb616c gen_send_ex
    @     0x7fd87e03d241 PyEval_EvalFrameEx
    @     0x7fd87dfb616c gen_send_ex
    @     0x7fd87e03d241 PyEval_EvalFrameEx
    @     0x7fd87dfb616c gen_send_ex
    @     0x7fd87e03d241 PyEval_EvalFrameEx
    @     0x7fd87dfb616c gen_send_ex
    @     0x7fd87e03d241 PyEval_EvalFrameEx
    @     0x7fd87dfb616c gen_send_ex
    @     0x7fd87e03d241 PyEval_EvalFrameEx
    @     0x7fd87dfb616c gen_send_ex
    @     0x7fd87e03d241 PyEval_EvalFrameEx
    @     0x7fd87dfb616c gen_send_ex
    @     0x7fd87e03d241 PyEval_EvalFrameEx
    @     0x7fd87e043bce PyEval_EvalCodeEx
    @     0x7fd87e04220a PyEval_EvalFrameEx
    @     0x7fd87e042560 PyEval_EvalFrameEx
    @     0x7fd87e043bce PyEval_EvalCodeEx
    @     0x7fd87e043ce2 PyEval_EvalCode
    @     0x7fd87e0639e0 PyRun_FileExFlags
    @     0x7fd87e063bbf PyRun_SimpleFileExFlags
    @     0x7fd87e079454 Py_Main
    @     0x7fd87d32dcdd __libc_start_main
/root/paddlejob/run.sh: line 307:   802 Aborted                 (core dumped) python train.py
*********************error messages********************
[/root/paddlejob/run.sh : 310] [start_paddle_trainer]
[ERROR]: execute user cmd failed, Sun Nov 11 03:09:36 CST 2018
~/paddlejob/workspace
end_hook start ...
~/paddlejob/workspace/env_run ~/paddlejob/workspace
~/paddlejob/workspace
data_clear start ...
rm /root/paddlejob/workspace/env_run/train_data
upload start ...
current path:/root/paddlejob/workspace
~/paddlejob/workspace ~/paddlejob/workspace
start upload model
upload model to hdfs
18/11/11 03:09:37 INFO fs.LibdfsLoader: Trying to load the libdfs library...
18/11/11 03:09:37 INFO fs.LibdfsLoader: Loaded the libdfs library
18/11/11 03:09:37 INFO fs.DFileSystem: Loaded the libdfs library
PADDLE_TRAINER_ID=0

**Comment by jacquesqiao** [comment]
请问您使用的是v2还是fluid？模型配置方便贴一下么

**Comment by liushanshan07** [comment]
> 请问您使用的是v2还是fluid？模型配置方便贴一下么

fluid。
--------tran.py
import paddle.fluid as fluid
from thirdparty.utility import add_arguments, print_arguments, to_lodtensor, get_feeder_data
from thirdparty.crnn_ctc_model_senet_5555_346_222_tf import ctc_train_net
import thirdparty.ctc_reader_2w as ctc_reader
import argparse
import functools
import sys
import time
import os
import numpy as np

parser = argparse.ArgumentParser(description=__doc__)
add_arg = functools.partial(add_arguments, argparser=parser)
# yapf: disable
add_arg('batch_size',        int,   32,         "Minibatch size.")
add_arg('total_step',        int,   720000*40,    "Number of training iterations.")
add_arg('log_period',        int,   1000,       "Log period.")
add_arg('save_model_period', int,   3000,      "Save model period. '-1' means never saving the model.")
add_arg('eval_period',       int,   3000,      "Evaluate period. '-1' means never evaluating the model.")
add_arg('save_model_dir',    str,   "./output/models", "The directory the model to be saved to.")
add_arg('init_model',        str,   "./thirdparty/2w_34.39/model_594000",       "The init model file of directory.")
add_arg('use_gpu',           bool,  True,      "Whether use GPU to train.")
add_arg('min_average_window',int,   10000,     "Min average window.")
add_arg('max_average_window',int,   15625,     "Max average window. It is proposed to be set as the number of minibatch in a pass.")
add_arg('average_window',    float, 0.15,      "Average window.")
add_arg('parallel',          bool,  True,     "Whether use parallel training.")
add_arg('l2',                float, 0.0004, "L2 regularizer.")
add_arg('learning_rate',     float, 1.0e-3, "Learning rate.")
add_arg('momentum',          float, 0.9,    "Momentum.")
# yapf: enable

def train(args, data_reader=ctc_reader):
    """OCR CTC training"""
    num_classes = None
    num_classes = data_reader.num_classes(
    ) if num_classes is None else num_classes
    data_shape = data_reader.data_shape()
    os.makedirs(args.save_model_dir)
    # define network
    images = fluid.layers.data(name='pixel', shape=data_shape, dtype='float32')
    label = fluid.layers.data(
        name='label', shape=[1], dtype='int32', lod_level=1)
    sum_cost, error_evaluator, inference_program, model_average = ctc_train_net(
        images, label, args, num_classes)

     # data reader
    train_reader = data_reader.train(args.batch_size)
    test_reader = data_reader.test(1)

   # prepare environment
    place = fluid.CPUPlace()
    if args.use_gpu:
        place = fluid.CUDAPlace(0)
    exe = fluid.Executor(place)
    exe.run(fluid.default_startup_program())

    # load init model
    if args.init_model is not None:
        model_dir = args.init_model
        model_file_name = None
        if not os.path.isdir(args.init_model):
            model_dir = os.path.dirname(args.init_model)
            model_file_name = os.path.basename(args.init_model)
        fluid.io.load_params(exe, dirname=model_dir, filename=model_file_name)
        print "Init model from: %s." % args.init_model

    train_exe = exe
    error_evaluator.reset(exe)
    if args.parallel:
        train_exe = fluid.ParallelExecutor(
            use_cuda=True, loss_name=sum_cost.name)

    fetch_vars = [sum_cost] + error_evaluator.metrics

    def train_one_batch(data):
        var_names = [var.name for var in fetch_vars]
        if args.parallel:
            results = train_exe.run(var_names,
                                    feed=get_feeder_data(data, place))
            results = [np.array(result).sum() for result in results]
        else:
            results = exe.run(feed=get_feeder_data(data, place),
                              fetch_list=fetch_vars)
            results = [result[0] for result in results]
        return results

    def test(iter_num):
        error_evaluator.reset(exe)
        for data in test_reader():
            exe.run(inference_program, feed=get_feeder_data(data, place))
        _, test_seq_error = error_evaluator.eval(exe)
        print "\nTime: %s; Iter[%d]; Test seq error: %s.\n" % (
            time.time(), iter_num, str(test_seq_error[0]))

    def save_model(args, exe, iter_num):
        filename = "model_%05d" % iter_num
        fluid.io.save_params(
            exe, dirname=args.save_model_dir, filename=filename)
        print "Saved model to: %s/%s." % (args.save_model_dir, filename)

    iter_num = 0
    while True:
        total_loss = 0.0
        total_seq_error = 0.0
        # train a pass
        for data in train_reader():
            iter_num += 1
            if iter_num > args.total_step:
                return
            results = train_one_batch(data)
            total_loss += results[0]
            total_seq_error += results[2]
            # training log
            if iter_num % args.log_period == 0:
                print "\nTime: %s; Iter[%d]; Avg Warp-CTC loss: %.3f; Avg seq err: %.3f" % (
                    time.time(), iter_num,
                    total_loss / (args.log_period * args.batch_size),
                    total_seq_error / (args.log_period * args.batch_size))
                sys.stdout.flush()
                total_loss = 0.0
                total_seq_error = 0.0

            # evaluate
            if iter_num % args.eval_period == 0:
                if model_average:
                    with model_average.apply(exe):
                        test(iter_num)
                else:
                    test(iter_num)

            # save model
            if iter_num % args.save_model_period == 0:
                if model_average:
                    with model_average.apply(exe):
                        save_model(args, exe, iter_num)
                else:
                    save_model(args, exe, iter_num)


def main():
    args = parser.parse_args()
    print_arguments(args)
    train(args, data_reader=ctc_reader)

if __name__ == "__main__":
    main()
--------------网络配置
regularizer = fluid.regularizer.L2Decay(args.l2)
352
353     fc_out = encoder_net(images, num_classes, regularizer=regularizer)
354     cost = fluid.layers.warpctc(
355         input=fc_out, label=label, blank=num_classes, norm_by_times=True)
356     sum_cost = fluid.layers.reduce_sum(cost)
357     decoded_out = fluid.layers.ctc_greedy_decoder(
358         input=fc_out, blank=num_classes)
359     casted_label = fluid.layers.cast(x=label, dtype='int64')
360     error_evaluator = fluid.evaluator.EditDistance(
361         input=decoded_out, label=casted_label)
362     inference_program = fluid.default_main_program().clone(for_test=True)
363     optimizer = fluid.optimizer.Momentum(
364         learning_rate=args.learning_rate, momentum=args.momentum)
365     _, params_grads = optimizer.minimize(sum_cost)
366     model_average = None
367     if args.average_window > 0:
368         model_average = fluid.optimizer.ModelAverage(
369             args.average_window,
370             min_average_window=args.min_average_window,
371             max_average_window=args.max_average_window)
372     return sum_cost, error_evaluator, inference_program, model_average
373

**Comment by jacquesqiao** [comment]
方便整理一下贴出来的代码的格式么，现在看着有点乱

**Comment by liushanshan07** [comment]
> 方便整理一下贴出来的代码的格式么，现在看着有点乱
我在paddlecloud上训练的，给你个作业链接吧。
作业详情链接：http://paddlecloud.baidu-int.com/paddle/jobInfo?jobId=job-e6c5be4f68768fc2&flag=jobs&groupName=k8s_gpu_demo&groupId=c0a1f165-6279-5320-b9e7-e0218c7a87f5&currentPage=1&currentKey=1
log链接：http://10.255.121.15:8388/?path=/home/work/containers_backup/369611898.job-e6c5be4f68768fc2-trainer_20181111_031050/home/crim/

**Comment by jacquesqiao** [comment]
时间比较长了，有问题线上直接联系，先关掉
